file_util.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2021 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+//lint:file-ignore U1000 Ignore all unused code
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/base64"
+	"hash"
+	"io"
+	"io/ioutil"
+	usr "os/user"
+	"path/filepath"
+	"strings"
+)
+
+type snowflakeFileUtil struct {
+}
+
+func (util *snowflakeFileUtil) compressFileWithGzipFromStream(srcStream **bytes.Buffer) (*bytes.Buffer, int) {
+	buf := getByteBufferContent(srcStream)
+	var c bytes.Buffer
+	w := gzip.NewWriter(&c)
+	w.Write(*buf) // write buf to gzip writer
+	w.Close()
+	return &c, c.Len()
+}
+
+func (util *snowflakeFileUtil) getDigestAndSize(src **bytes.Buffer) (string, int64) {
+	useOpenSSL := getUseOpenSSL()
+	var m hash.Hash
+	if !useOpenSSL {
```

Comment:
seems if useOpenSSL is true, digest is not computed at all so this method could directly return nil, src.len